wled00/json.cpp

```diff
@@ -486,7 +486,11 @@ bool deserializeState(JsonObject root, byte callMode, byte presetId)
       strip.loadCustomPalettes();
     }
   }
-
+  
+  if (root.containsKey(F("np")) && root[F("np")].as<bool>()) { //skip to next preset in a playlist
+    handlePlaylist(true);
```

Comment:
On esp32, this call to handlePlaylist() would run in `awync_tcp` context and not in arduino context. It might be dangerous because handlePlaylist can replace segments while effects are drawing.

I think it would be safer to set a boolean flag (like `doAdvancePlaylist)`, and use this as parameter in the main  wled::loop(). This way we can be sure that effects are not drawing while segments get deleted.

Reply:
Cool, thanks good feedback. I reworked it with your approach, it works fine in my testing.